src/controllers/lyrics.js

```diff
@@ -0,0 +1,239 @@
+        }
+    }
+
+    function renderDynamicLyrics(lyrics) {
+        const itemsContainer = view.querySelector('.dynamicLyricsContainer');
+        if (itemsContainer) {
+            const html = lyrics.reduce(dynamicLyricHtmlReducer, '');
+            itemsContainer.innerHTML = html;
+        }
+        const lyricLineArray = itemsContainer.querySelectorAll('.lyrics');
+
+        // attaches click event listener to change playtime to lyric start
+        lyricLineArray.forEach(element => {
+            element.addEventListener('click', () => onLyricClick(element.getAttribute('data-lyrictime')));
+        });
+
+        const currentIndex = getLyricIndex(getCurrentPlayTime(), lyrics);
+        updateAllLyricLines(currentIndex, savedLyrics);
+    }
+
+    function renderStaticLyrics(lyrics) {
+        const itemsContainer = view.querySelector('.dynamicLyricsContainer');
+        if (itemsContainer) {
+            const html = lyrics.reduce(staticLyricHtmlReducer, '');
+            itemsContainer.innerHTML = html;
+        }
+    }
+
+    function updateLyrics(lyrics) {
+        savedLyrics = lyrics;
+
+        isDynamicLyric = Object.prototype.hasOwnProperty.call(lyrics[0], 'Start');
+
+        if (isDynamicLyric) {
+            renderDynamicLyrics(savedLyrics);
+        } else {
+            renderStaticLyrics(savedLyrics);
+        }
+    }
+
+    function getLyrics (serverId, itemId) {
+        const apiClient = ServerConnections.getApiClient(serverId);
+
+        return apiClient.ajax({
+            url: apiClient.getUrl('Audio/' + itemId + '/Lyrics'),
+            type: 'GET',
+            dataType: 'json'
+        }).then((response) => {
+            if (!response.Lyrics) {
+                throw new Error();
+            }
+            return response.Lyrics;
+        });
+    }
+
+    function bindToPlayer (player) {
+        if (player === currentPlayer) {
+            return;
+        }
+
+        releaseCurrentPlayer();
+
+        currentPlayer = player;
+
+        if (!player) {
+            return;
+        }
+
+        Events.on(player, 'timeupdate', onTimeUpdate);
+        Events.on(player, 'playbackstart', onPlaybackStart);
+        Events.on(player, 'playbackstop', onPlaybackStop);
+    }
+
+    function releaseCurrentPlayer() {
+        const player = currentPlayer;
+
+        if (player) {
+            Events.off(player, 'timeupdate', onTimeUpdate);
+            Events.off(player, 'playbackstart', onPlaybackStart);
+            Events.off(player, 'playbackstop', onPlaybackStop);
+            currentPlayer = null;
+        }
+    }
+
+    function onLyricClick(lyricTime) {
+        playbackManager.seek(lyricTime);
+        if (playbackManager.paused()) {
+            playbackManager.playPause(currentPlayer);
+        }
+    }
+
+    function onTimeUpdate(_, ticks) {
+        ticks *= TICKS_PER_SECOND;
+        if (isDynamicLyric) {
+            const currentIndex = getLyricIndex(ticks, savedLyrics);
```

Comment:
What's the problem with `getCurrentPlayTime()`?
I don't think we should modify the `timeupdate` event for no reason. _In fact, it has data that differs from player to player - not good, imo._